### PR TITLE
closes #18: Fix Export fails if folder already exists

### LINKED ISSUE #18: Export fails if the folder already exists
* folder where assets are added should have random name
* it will anyway be deleted at the end, and ZIP only contains the folder data and not the folder itself

image

### PR DESCRIPTION
**Pull request by vladavoX**
Original issue: #18

### CODE REVIEW

File: src/node/export.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ const fs = require('fs');
 const fsPromises = require('fs/promises');
 const archiver = require('archiver');
 const path = require('path');
+const crypto = require('crypto');
```
Comment (ivansenic) [inline src/node/export.ts:5]: why was this added?

File: src/node/export.ts [hunk 2]
```diff
@@ -119,11 +121,15 @@ async function copyFootage(footage: Footage[], projectDir: string) {
...
  * @param zipPath The path of the directory to zip.
  */
-function zip(zipPath: string): Promise<void> {
+function zip(zipPath: string, projectName: string): Promise<void> {
   return new Promise((resolve, reject) => {
     const pathResolved = finalizePath(zipPath); // Normalize and resolve
 
-    const outputZipPath = `${pathResolved}.zip`;
+    // replace final part of path with project name to create zip with project name
+    const pathReplaced = path.join(path.dirname(pathResolved), projectName);
```
Comment (ivansenic) [inline src/node/export.ts:129]: what's the meaning of the `pathReplaced`? even the comment above is completely unclear..

File: src/node/export.ts [hunk 3]
```diff
@@ -33,24 +34,25 @@ function selectFolder(callback: (result: string) => void) {
  * @param targetPath the path of the folder where the files will be copied.
  * @param callback a callback that will be called with the project name.
  */
-async function collectFiles(targetPath: string): Promise<string> {
+async function collectFiles(targetPath: string): Promise<[string, string]> {
```
Comment (ivansenic) [inline src/node/export.ts:37]: Is this the best way of returning a result? What would be wrong with the type
```
{
   collectedFilesDir: string;
   projectName: string;
}
```

File: src/node/export.ts [hunk 4]
```diff
@@ -119,11 +121,15 @@ async function copyFootage(footage: Footage[], projectDir: string) {
  *
  * @param zipPath The path of the directory to zip.
  */
-function zip(zipPath: string): Promise<void> {
+function zip(zipPath: string, projectName: string): Promise<void> {
```
Comment (ivansenic) [inline src/node/export.ts:124]: * `zipPath` is also wrong, should be `targetPath` or something like that
* can you please go over naming here so other people would know what's going on when they are looking at the code

File: src/node/export.ts [hunk 5]
```diff
@@ -33,24 +34,25 @@ function selectFolder(callback: (result: string) => void) {
...
   await fsPromises.copyFile(finalizePath(projectInfo.projectPath), dest);
 
   await copyFonts(projectInfo.fonts, projectDir);
   await copyFootage(projectInfo.footage, projectDir);
 
-  return projectDir;
+  return [projectDir, projectName];
 }
 
 async function copyFonts(fonts: Fonts[], projectDir: string) {
```
Comment (ivansenic) [inline src/node/export.ts:58]: also `targetDir` here..

File: src/components/form/ExportForm.tsx [hunk 6]
```diff
@@ -27,8 +27,8 @@ export default function ExportForm() {
     setLoading(true);
     if (targetPath) {
       try {
-        const result = await collectFiles(targetPath);
-        await zip(result);
+        const [result, projectName] = await collectFiles(targetPath);
```
Comment (ivansenic) [inline src/components/form/ExportForm.tsx:30]: I get a wrong notification

image

**Review by ivansenic — CHANGES_REQUESTED** [review]

File: src/node/export.ts (hunk 1 above)
Comment (vladavoX) [reply src/node/export.ts:5]: so it generates random uuid, crypto is nodes module

File: src/node/export.ts [hunk 7]
```diff
@@ -114,16 +115,20 @@ async function copyFootage(footage: Footage[], projectDir: string) {
...
   return new Promise((resolve, reject) => {
-    const pathResolved = finalizePath(zipPath); // Normalize and resolve
+    const targetPathResolved = finalizePath(targetPath); // Normalize and resolve
 
-    const outputZipPath = `${pathResolved}.zip`;
+    // replace final part of path (random uuid) with project name to create zip with projectName.zip
+    const zipPath = path.join(path.dirname(targetPathResolved), projectName);
+    const finalizeZipPath = finalizePath(zipPath); // Normalize and resolve
+
+    const outputZipPath = `${finalizeZipPath}.zip`;
```
Comment (ivansenic) [inline src/node/export.ts:131]: imo extension should already be there, right? same like on line `41`

```
    // replace final part of path (random uuid) with project name to create zip with projectName.zip
    const zipPath = path.join(path.dirname(targetPathResolved), `${projectName}.zip`);
    const outputZipPath = finalizePath(zipPath); // Normalize and resolve
```

**Review by ivansenic — APPROVED** [review]
Can you please test once again before merging, it takes too much time for me to start the dockur..

**Comment by vladavoX** [comment]
### When it exists

image
image

### when it doesn't, and also notification message
image

File: script/updateManifest.cjs [hunk 8]
```diff
@@ -37,8 +37,8 @@ fs.readFile(manifestPath, 'utf-8', (err, data) => {
     .replace(
       /<Menu>(.*?)<\/Menu>/,
       isDev
-        ? '<Menu>Plainly plugin - Dev mode</Menu>'
-        : '<Menu>Plainly plugin</Menu>',
+        ? '<Menu>Plainly Videos - Dev mode</Menu>'
+        : '<Menu>Plainly Videos</Menu>',
```
Comment (vladavoX) [inline script/updateManifest.cjs:41]: @ivansenic from your PR, needs to be changed here also, so `yarn build/dev` produce the same name

**Pull request merged**